src/test/java/uk/gov/pay/ledger/event/resource/EventResourceAgreementIT.java

```diff
@@ -85,4 +76,45 @@ public void shouldWriteCancelledEvent() {
         assertThat(agreementEntity.getCancelledByUserEmail(), is("jdoe@example.org"));
         assertThat(agreementEntity.getCancelledDate(), is(ZonedDateTimeMatchers.within(1, ChronoUnit.SECONDS, now)));
     }
+
+    @Test
+    public void shouldNotWriteEventIfAMandatoryFieldIsNotPresent() {
+        var now = ZonedDateTime.now(ZoneOffset.UTC);
+
+        insertAgreementCreatedEventFixture();
+
+        var params = new JSONArray();
+        //event missing eventType
+        var event = new JSONObject()
+                .put("service_id", serviceId)
+                .put("resource_type", "agreement")
+                .put("live", false)
+                .put("timestamp", now.toString())
+                .put("event_details", new JSONObject()
+                        .put("user_email", "jdoe@example.org")
+                        .put("cancelled_date", now.toString()))
+                .put("resource_external_id", agreementId);
+        params.put(event);
+
+        given().port(port)
+                .contentType(JSON)
+                .body(params.toString())
+                .post("/v1/event")
+                .then()
+                .statusCode(422);
```

Comment:
I think testing 422 here is fine but not clear which field is supposed to be mandatory as per the test name. 

Would be good to update the test name or add a few more assertions for the expected fields without relying on the comment.